src/Microsoft.Azure.WebJobs.Host/FunctionInvocationException.cs

```diff
@@ -18,12 +20,13 @@ public FunctionInvocationException() : base()
         }
 
         /// <inheritdoc/>
-        public FunctionInvocationException(string message) : base(message)
+        public FunctionInvocationException(string message) : base(Sanitizer.Sanitize(message))
         {
         }
 
         /// <inheritdoc/>
-        public FunctionInvocationException(string message, Exception innerException) : base(message, innerException)
+        public FunctionInvocationException(string message, Exception innerException)
+            : base(Sanitizer.Sanitize(message), UpdateMessage(innerException, Sanitizer.Sanitize(innerException?.Message)))
```

Comment:
I think that we should be doing the Sanitization where the log is actually written, not when the exception is constructed. For example, we want to retain the ability of .NET Isolated custom middleware to receive the full exception info as thrown by the customer code.

So we should trace the handling/logging of this exception up to where it's actually written. Basically anywhere where we're handling and logging exception details for exceptions that come from user code, we need to sanitize. That's beyond just function invocations, but also custom startup code as well, which you can verify by throwing a similar exception from custom startup logic.

In the Azure Functions Host, we actually attempt to sanitize exceptions [here](https://github.com/Azure/azure-functions-host/blob/aa332f6c72c0b87ec83dec5bb7b9ac24d9947229/src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs#L181) where they're logged. Why isn't that logic handling this? Or perhaps we need to update that code to also sanitize innerExceptionMessage?

Reply:
I will verify why that logic is not handling this and get back on the PR.